config/src/test/java/org/axonframework/config/SimpleStatefulCommandHandlingComponentTest.java

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.config;
+
+
+import org.axonframework.commandhandling.GenericCommandMessage;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.eventsourcing.AnnotationBasedEventStateApplier;
+import org.axonframework.eventsourcing.AsyncEventSourcingRepository;
+import org.axonframework.eventsourcing.EventSourcingHandler;
+import org.axonframework.eventsourcing.EventStateApplier;
+import org.axonframework.eventsourcing.annotations.EventTag;
+import org.axonframework.eventsourcing.eventstore.AnnotationBasedTagResolver;
+import org.axonframework.eventsourcing.eventstore.EventCriteria;
+import org.axonframework.eventsourcing.eventstore.SimpleEventStore;
+import org.axonframework.eventsourcing.eventstore.Tag;
+import org.axonframework.eventsourcing.eventstore.inmemory.AsyncInMemoryEventStorageEngine;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.MessageType;
+import org.axonframework.messaging.QualifiedName;
+import org.axonframework.messaging.unitofwork.AsyncUnitOfWork;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.axonframework.modelling.command.ModelRegistry;
+import org.axonframework.modelling.command.SimpleModelRegistry;
+import org.axonframework.modelling.command.StatefulCommandHandlingComponent;
+import org.axonframework.modelling.repository.ManagedEntity;
+import org.junit.jupiter.api.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Tests the fluent interface for the configuration of a {@link StatefulCommandHandlingComponent}.
+ */
+class SimpleStatefulCommandHandlingComponentTest {
+
+    public static final String DEFAULT_CONTEXT = "default";
+
+    private final SimpleEventStore eventStore = new SimpleEventStore(
+            new AsyncInMemoryEventStorageEngine(),
+            DEFAULT_CONTEXT,
+            new AnnotationBasedTagResolver()
+    );
+
+    private final EventStateApplier<Student> studentEventStateApplier = new AnnotationBasedEventStateApplier<>(Student.class);
+    private final EventStateApplier<Course> courseEventStateApplier = new AnnotationBasedEventStateApplier<>(Course.class);
```

Comment:
To consider: 
I like you tested if everything works together, but I'm not sure if it should be part of this test or we should have both - annotation based and what you had before: inline implementation for the test purpose.
Or maybe make a TestSuite for that? Then we will be able to test it with other EventStateApplier implementations and we will be sure what broke the test (in case of bug in `AnnotationBasedEventStateApplier` only test suite with that will fail).

Reply:
It's a nice idea, I will apply it.
Although I do have to say that this test is more for demonstration purposes, and I expect it to be ported one day to another module in another form.

Reply:
The new `AbstractStudentTestSuite` uses a mix of the two by default :) As such, resolving this comment